### PR TITLE
Feat/tmi2 611

### PR DESCRIPTION
**Pull request by dylanwrightCO**
## Description

Ticket # and link

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

**Review by GavCookCO — APPROVED** [review]

File: .github/workflows/promoteToProd.yml [hunk 1]
```diff
@@ -14,14 +14,17 @@ jobs:
...
-        uses: aws-actions/configure-aws-credentials@v1-node16
+      - uses: aws-actions/configure-aws-credentials@v3
+        name: Configure AWS credentials
         with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: github-gap-automated-tests
+          aws-region: ${{ vars.AWS_REGION }}
```
Comment (dominicwest) [inline .github/workflows/promoteToProd.yml:27]:
```suggestion
          aws-region: ${{ env.AWS_REGION }}
```

File: .github/workflows/pushImage.yml [hunk 2]
```diff
@@ -72,12 +75,12 @@ jobs:
...
-        uses: aws-actions/configure-aws-credentials@v1-node16
+      - uses: aws-actions/configure-aws-credentials@v3
+        name: Configure AWS credentials
         with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: github-gap-automated-tests
+          aws-region: ${{ vars.AWS_REGION }}
```
Comment (dominicwest) [inline .github/workflows/pushImage.yml:83]:
```suggestion
          aws-region: ${{ env.AWS_REGION }}
```

File: .github/workflows/pushImage.yml [hunk 3]
```diff
@@ -126,12 +132,12 @@ jobs:
...
```
Comment (dominicwest) [inline .github/workflows/pushImage.yml:140]:
```suggestion
          aws-region: ${{ env.AWS_REGION }}
```

File: src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java [hunk 4]
```diff
@@ -216,6 +220,16 @@ public User updateRoles(Integer id, List<Integer> newRoles) {
         return user;
     }
 
+    private void handleTechSupportRoleChange(User user, UpdateUserRolesRequestDto updateUserRolesRequestDto, String jwt) {
+        if (updateUserRolesRequestDto.newUserRoles().contains(5) && !user.isTechnicalSupport()) {
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java:224]: Ideally lets look up `5` from the enum for readability

File: src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java [hunk 5]
```diff
@@ -216,6 +220,16 @@ public User updateRoles(Integer id, List<Integer> newRoles) {
...
+            addTechSupportUserToApply(user, updateUserRolesRequestDto.departmentId(), jwt);
+        } else {
+            if (user.isTechnicalSupport() && !updateUserRolesRequestDto.newUserRoles().contains(5)) {
```
Comment (dominicwest) [inline src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java:227]: as above

**Review by dominicwest — CHANGES_REQUESTED** [review]

**Review by dominicwest — APPROVED** [review]

**Review by GavCookCO — APPROVED** [review]

**Pull request merged**